synfig-studio/src/gui/app.cpp

```diff
@@ -3614,7 +3614,10 @@ App::dialog_message_2b(const std::string &message,
 	dialog.add_button(button1, 0);
 	dialog.add_button(button2, 1);
 
-	return	dialog.run();
+	if(dialog.run() == 1)
```

Comment:
Are you sure 1 should be the correct/OK button?

Anyway Gtk::RESPONSE_NONE (= -1) is expected when user closes dialog, I guess

Reply:
In practice it works... but I don't remember where I read it in the documentation.
I think the easiest way is to "catch" the event where a positive response is given (only one case) and should be 1, because there are many cases like the one you reported or Gtk::RESPONSE_DELETE_EVENT.

Anyway I'll check it later and reply here.

Reply:
@rodolforg I checked and it's correct.
The returned id value is defined with the button here as 1:
https://github.com/synfig/synfig/blob/6e3297813bc4f825a9019c837be420857f65cd21/synfig-studio/src/gui/app.cpp#L3615

Reply:
Nice! Approved then ;)